.pre-commit-config.yaml

```diff
@@ -78,6 +78,26 @@ repos:
           ?^cpp/src/generated/|
           ?^cpp/thirdparty/|
           )
+  - repo: https://github.com/cpplint/cpplint
+    rev: 1.6.1
+    hooks:
+      - id: cpplint
+        name: C++ Lint
+        args:
+          - "--verbose=2"
+        types_or:
+          - c++
+        files: >-
+          ^cpp/
+        exclude: >-
+          (
+          ?\.grpc\.fb\.(cc|h)$|
```

Comment:
The only nit is that this does not match the exclusion list that we have today in: https://github.com/apache/arrow/blob/main/cpp/build-support/lint_exclusions.txt
I can see some `pyarrow` files there. We can merge this PR and add a follow up issue to add C++ linting to pre-commit to python and R (which seems to be what was originally linted)

Reply:
Good catch! I missed the file.
It seems that `*windows_compatibility.h` difference is only affected for `cpp/`.
But we can use cpplint for *windows_compatibility.h` now:

* https://github.com/apache/arrow/blob/main/cpp/src/arrow/util/windows_compatibility.h
* https://github.com/apache/arrow/blob/main/cpp/src/parquet/windows_compatibility.h

So this list is valid.

I also missed that R also uses cpplint: https://github.com/apache/arrow/blob/main/r/lint.sh
Let's work on it as a separated PR.

It seems that the current Python code base doesn't use cpplint.

BTW, I doubt that we may not need cpplint: https://github.com/apache/arrow/issues/37396